client.go

```diff
@@ -114,6 +115,14 @@ func WithHTTPHost(scheme, host string) Option {
 	}
 }
 
+// WithHTTPHeaders returns an option for setting a custom HTTP headers.
+// These headers are set in addition to the ones defined in Client.setHeaders().
+// If a custom header is added with the same key as one of default header, then
+// custom value is appended to key, and does not replace default value.
+func WithHTTPHeaders(headers map[string]string) Option {
+	return func(c *Client) { c.customHeaders = headers }
```

Comment:
Would be a good idea to copy here. Not copying could lead to race conditions if the caller isn't aware that it's used as is. Another option is to accept key-value pairs:

```
func WithHTTPHeader(key, value string) Option
```

Internally, I think we should leverage `http.Header`.